openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -100,6 +101,11 @@ def _load_usernames(self):
             [Q(course_id=course_run.key, mode=course_run.type) for course_run in self.course_runs]
         )
 
+        # Account for the fact that no-id-professional and professional are equivalent
+        for course_run in self.course_runs:
```

Comment:
Not a blocker, but it's worth keeping an eye on what this duplicated loop does to the command's runtime. If it becomes a concern it can be combined with the `reduce` call above.